Merge of dev_sim2root_merge__merge_with_dev_fix_fields into dev

lwpiotr opened the pull request.

Commit:
New scripts to archive grand raw files

Commit:
New scripts to archive grand raw files

Commit:
New scripts to archive grand raw files

Commit:
New scripts to archive grand raw files

Commit:
New scripts to archive grand raw files

Commit:
New scripts to archive grand raw files

Commit:
Script createAIP.jar modified to work with java version > 8

Commit:
Script createAIP.jar modified to work with java version > 8

Commit:
Script createAIP.jar modified to work with java version > 8

Commit:
Script createAIP.jar modified to work with java version > 8

Commit:
Script createAIP.jar modified to work with java version > 8

Commit:
Script createAIP.jar modified to work with java version > 8

ifleg committed:
Delete scripts/archiving/create_archive.bash

Old file replaced by archive_grandraw.bash

Commit:
Unified script to be distributed on all sites.
Include a purge of the already transfered files from the database before transfer.

Commit:
Unified script to be distributed on all sites.
Include a purge of the already transfered files from the database before transfer.

Commit:
Unified script to be distributed on all sites.
Include a purge of the already transfered files from the database before transfer.

Commit:
Unified script to be distributed on all sites.
Include a purge of the already transfered files from the database before transfer.

Commit:
New version to manage gp80 data

Commit:
New version to manage gp80 data

Commit:
New version to manage gp80 data

Commit:
Corrected small bug to get the correct python3 path in bash scripts

Commit:
Corrected small bug to get the correct python3 path in bash scripts

Commit:
Added lock feature to prevent launching the script if a previous instance is running

Commit:
Added lock feature to prevent launching the script if a previous instance is running

Commit:
Correction of gtot options

Commit:
Correction of gtot options

Commit:
Modifications to manage datasets. Not fully tested yet !

Commit:
Modifications to avoid to run too many jobs at the same time (which can saturate the database).

Commit:
Modifications to avoid to run too many jobs at the same time (which can saturate the database).

Commit:
Modifications to avoid to run too many jobs at the same time (which can saturate the database).

Commit:
Modifications to avoid to run too many jobs at the same time (which can saturate the database).

Commit:
Add info for archiving GP80 files

Commit:
Change in paths for prod @ccin2p3

Commit:
Change in paths for prod @ccin2p3

Commit:
Use of overwrite option for conversion

Commit:
Use of overwrite option for conversion

Commit:
Corrected bugs related to the definition of config file.
Removed el9 selection.
Added overwrite option for registering into the database

Commit:
Corrected bugs related to the definition of config file.
Removed el9 selection.
Added overwrite option for registering into the database

Commit:
Manage output status of bash scripts in case of errors in pythons scripts.

Commit:
Manage output status of bash scripts in case of errors in pythons scripts.

Commit:
Removed all import from grand/__init__.py and adjusted imports in other files when needed

Commit:
removed -l option when not necessary in bash files. Changed parameters of grouping in ccscript.bash

Commit:
removed -l option when not necessary in bash files. Changed parameters of grouping in ccscript.bash

Commit:
removed unecessary imports.

lwpiotr committed:
Merge branch 'dev_sim2root' into dev_sim2root_merge

lwpiotr committed:
If not available, filling energy_em with [0]

lwpiotr committed:
Merge branch 'dev_sim2root' into dev_sim2root_merge

lwpiotr committed:
Added a mandatory type conversion of numpy array to the vector type in StdVectorList iadd()

lwpiotr committed:
Merge branch 'dev_io_root' into dev_sim2root_merge

lwpiotr committed:
Merge remote-tracking branch 'origin/dev' into dev_sim2root_merge__merge_with_dev

# Conflicts:
#	data/download_data_grand.py
#	grand/sim/detector/rf_chain.py
#	grand/sim/efield2voltage.py
#	scripts/convert_efield2voltage.py
#	sim2root/Common/IllustrateSimPipe.py

lwpiotr committed:
Updated trunrawvoltage fields

lwpiotr committed:
Updated trunrawvoltage fields

mjtueros committed:
changed how to call the sim2root modules relative to GRAND_ROOT

lwpiotr committed:
Writing first/last_event(_time) into trun

lwpiotr committed:
Writing first/last_event(_time) into trun

lwpiotr committed:
Fixed start event number in filenames and trun

lwpiotr committed:
Fixed start event number in filenames and trun

lwpiotr committed:
Chanced trun.data_source to "simulation"

lwpiotr committed:
Chanced trun.data_source to "simulation"

lwpiotr committed:
Fixed some typos

lwpiotr committed:
Fixed some typos

lwpiotr committed:
Splitting trawshower.sim_name into trun.sim_name and sim_version

lwpiotr committed:
Splitting trawshower.sim_name into trun.sim_name and sim_version

lwpiotr committed:
Filling trun.data_generator

lwpiotr committed:
Filling trun.data_generator

lwpiotr committed:
GRANDCS now works with x/y/z arrays

lwpiotr committed:
GRANDCS now works with x/y/z arrays

lwpiotr committed:
Filling du_geoid in trun

lwpiotr committed:
Filling du_geoid in trun

lwpiotr committed:
Fixed trun.data_generator for coreas

lwpiotr committed:
Fixed trun.data_generator for coreas

Commit:
rebased on dev_io_root

Commit:
rebased on dev_io_root

Commit:
Modification on repository access

lwpiotr committed:
Fixed setting exp/sim structure in DataDirectory

lwpiotr committed:
Fixed setting exp/sim structure in DataDirectory

lwpiotr committed:
Merge branch 'dev_io_root' into dev_sim2root_merge__merge_with_dev_fix_fields

Commit:
Renamed granddatalib into datamanager

Commit:
Merge remote-tracking branch 'origin/dev_io_root' into dev_database

Commit:
Renamed granddatalib into datamanager

lwpiotr committed:
Properly filling initialised vector from vector now

lwpiotr committed:
Properly filling initialised vector from vector now

lwpiotr committed:
Filling event event_number and run_number

lwpiotr committed:
Filling event event_number and run_number

lwpiotr committed:
Added event's t_vector and getting all DUs votlage/efield at a specific time to Event

lwpiotr committed:
Added event's t_vector and getting all DUs votlage/efield at a specific time to Event

lwpiotr committed:
Added hilbert envelopes and getting them at specific times; Changed filling t_vector to 1 ns resolution

lwpiotr committed:
Added hilbert envelopes and getting them at specific times; Changed filling t_vector to 1 ns resolution

snonis committed:
Updated rf_chain.py to use XML-based configuration. Replaced hardcoded file paths with dynamic loading from rf_chain_config.xml. Improved flexibility, error handling, and axis-based file selection

lwpiotr committed:
Merge branch 'dev_io_root' into dev_sim2root_merge__merge_with_dev_fix_fields

snonis committed:
Updated rf_chain.py to use XML-based configuration. Replaced hardcoded file paths with dynamic loading from rf_chain_config.xml. Improved flexibility, error handling, and axis-based file selection.

lwpiotr committed:
Added get_current_file(); Building indices for TChains now.

lwpiotr committed:
Added get_current_file(); Building indices for TChains now.

lwpiotr committed:
Added get_current_file(); Building indices for TChains now.

lwpiotr committed:
Saving voltages to several files if efields are in several files; some fixes

lwpiotr committed:
Saving voltages to several files if efields are in several files; some fixes

lwpiotr committed:
Saving adcs to several files if voltages are in several files

lwpiotr committed:
Saving efields L1 to several files if efield L0 are in several files

lwpiotr committed:
Made executable

lwpiotr committed:
Changed rf chain config file path from absolute into resolved

mjtueros committed:
Added the shower electromagnetic energy to the ZHAireS rawroot file

Commit:
improved casttodb.
added register_in_db
added trunrawvoltage

Commit:
Updated grandlib_amd64 env and added a recipie to build a grandlib conda env from scratch

lwpiotr committed:
Merge remote-tracking branch 'origin/dev_sim2root_merge__merge_with_dev_fix_fields' into dev_sim2root_merge__merge_with_dev_fix_fields

mjtueros committed:
fixed shower Xmax position coordinates bug found by JeeanMarc.
Now it is expresed from ground level instad of sea level

Commit:
Remove from local db files older than 2 months and not yet transfered (because after 2 monthe it's not a transfer problem but more likely that files has been removed or unproperly registered before).

lwpiotr committed:
A script that opens GRAND events data as EventList from AOI

lwpiotr committed:
Example update

lwpiotr committed:
Output update

lwpiotr committed:
Merge branch 'dev_io_root' into dev_sim2root_merge__merge_with_dev_fix_fields

lwpiotr committed:
Added sim_version to trunshowersim and trunefieldsim; Changed efield_sim to sim_name in trunefieldsim

lwpiotr committed:
Some comments updates

lwpiotr committed:
Handling trunefieldsim and trunshowersim sim_name/sim_version

lwpiotr committed:
Added for proper imports

lwpiotr committed:
Fixed the imports

mjtueros committed:
Minor fix to bug introduced in last commit

jelenakhlr committed:
resolves issue #112
The Option/ArgeParse command line arguments now includes the specification of which parameters are mandatory. This way if the user will not give required parameters, the script will fail displaying proper information.

Commit:
Corrected bug on registering repository (due to change of database structure)

Commit:
Modification to handle destination of converted files (-o file.root if -os option and -od directory if new structure used)

lwpiotr committed:
Added site to trawefield. Changed default site to "" for trawshower.

lwpiotr committed:
Filling the site in trunshowersim and trunefieldsim

lwpiotr committed:
Added the site to trunefieldsim and changed comments for site in trunshowersim

lwpiotr committed:
Added the site to trunefieldsim and changed comments for site in trunshowersim

Commit:
corrected bug on registering file (id_repository was missing)

Commit:
Management of directories

lwpiotr committed:
Requiring site_layout command line parameter if not star shape. Setting trun.site_layout to this parameter if provided.

lwpiotr committed:
Requiring site_layout command line parameter if not star shape. Setting trun.site_layout to this parameter if provided.

mjtueros committed:
added sim_name, sim_version and site on trawshoer and trawefield

mjtueros committed:
added sim_name, sim_version and site on trawshoer and trawefield

Commit:
Modifications to avoid duplicate path in repositories

lguelzow committed:
fix bug in coreas version function that led to it being overwritten by "n/a"

lguelzow committed:
fix bug in coreas version function that led to it being overwritten by "n/a"

lguelzow committed:
new function to read Corsika version from .inp file

lguelzow committed:
new function to read Corsika version from .inp file

lguelzow committed:
read corsika version and add shower and efield generator and version to new fields

lguelzow committed:
read corsika version and add shower and efield generator and version to new fields

lguelzow committed:
fix bug that lead to correct version of hadr. model to be overwritten

lwpiotr committed:
Added one writeout

lwpiotr committed:
Split root_trees.py into several files

lwpiotr committed:
Adjusted the code to the root_trees.py split into several files

lwpiotr committed:
Merge remote-tracking branch 'origin/dev_io_root' into dev_io_root

# Conflicts:
#	grand/dataio/root_trees.py

lwpiotr committed:
Deleting root_trees.py

lwpiotr committed:
Improved the paths and output

jelenakhlr committed:
Update CorsikaInfoFuncs.py

make reading the corsika version more reliable and future-proof

lwpiotr committed:
Split and refactoring of grandlib_classes.py to aoi

jelenakhlr committed:
handle missing version numbers (in corsika 7)

jelenakhlr committed:
update readme to the actual current example files

jelenakhlr committed:
remove "coreas" and "corsika" from the version numbers - the names are already in the sim_name fields

jelenakhlr committed:
remove hardcoded Coreas too

lwpiotr committed:
Fixed to import correct dataio after the split

lwpiotr committed:
No fail, just warning in case a tree tries to read a non-existing branch

snonis committed:
new files for the antenna effective length and new precomputed file for the input of the galactic noise.

snonis committed:
A small modification in grand/data/readme.md explaining how to download the new antenna model.

lwpiotr committed:
Merge branch 'dev_snonis' into dev_sim2root_merge__merge_with_dev_fix_fields

# Conflicts:
#	data/download_data_grand.py
#	grand/sim/detector/rf_chain.py

lwpiotr committed:
No fail, just warning in case a tree tries to read a non-existing branch

lwpiotr committed:
Merge remote-tracking branch 'origin/dev' into dev_sim2root_merge__merge_with_dev_fix_fields

jelenakhlr committed:
Delete sim2root/CoREASRawRoot/coreas_pipeline.py

This pipeline is not up-to-date anymore

lguelzow committed:
Merge branch 'dev_sim2root_merge__merge_with_dev_fix_fields' of https://github.com/grand-mother/grand into dev_sim2root_merge__merge_with_dev_fix_fields

ifleg committed:
Update readme.md

Corrected path to dataio

lwpiotr committed:
Merge remote-tracking branch 'origin/dev_sim2root_merge__merge_with_dev_fix_fields' into dev_sim2root_merge__merge_with_dev_fix_fields

lwpiotr committed:
Added the readme for AOI examples

lwpiotr committed:
Remove failing on missing branch for ROOT.string

lwpiotr committed:
Merge branch 'dev_io_root' into dev_sim2root_merge__merge_with_dev_fix_fields

mjtueros committed:
minnor bugfixing

mjtueros committed:
Added memmory usage output to screee in voltage2adc script

lwpiotr committed:
Merge remote-tracking branch 'origin/dev_sim2root_merge__merge_with_dev_fix_fields' into dev_sim2root_merge__merge_with_dev_fix_fields

lwpiotr committed:
self.__setattr__ is now pointing to self.mod_setattr through weak reference to avoid troubles with cleanup

lwpiotr committed:
Do not fill/modify vectors if empty value was given

lwpiotr committed:
Filling entry_list only if it is empty

lwpiotr committed:
Merge branch 'dev_io_root' into dev_sim2root_merge__merge_with_dev_fix_fields

lguelzow committed:
Merge branch 'dev_sim2root_merge__merge_with_dev_fix_fields' of https://github.com/grand-mother/grand into dev_sim2root_merge__merge_with_dev_fix_fields

lwpiotr committed:
Merge remote-tracking branch 'origin/dev_sim2root_merge__merge_with_dev_fix_fields' into dev_sim2root_merge__merge_with_dev_fix_fields

Commit:
Updated with choice of month before current date to archive

lguelzow committed:
change ADC and L1 efield trace length to 2048ns to match length of data files. Shift trigger position to 550 ns

lguelzow committed:
Merge branch 'dev_sim2root_merge__merge_with_dev_fix_fields' of https://github.com/grand-mother/grand into dev_sim2root_merge__merge_with_dev_fix_fields

lguelzow committed:
typo in L1 efield trace length

Commit:
Automated procedure and scripts to backup raw files

lguelzow committed:
add calculation of Xmax cart. position to rawroot

lguelzow committed:
Merge branch 'dev_sim2root_merge__merge_with_dev_fix_fields' of https://github.com/grand-mother/grand into dev_sim2root_merge__merge_with_dev_fix_fields

snonis committed:
1. Update download_data_grand.py to handle new data package version via flag file check and automatic re-download if missing. 2. Update notebooks examples/sim/plot_VoltageAtDevice.ipynb, examples/sim/rf_chain_example.ipynb, examples/read_modify_RF_chain_elements.ipynb, based on reviewer's feedback 3. Minor change in script galaxy.py regarding frequency range in galactic models with nec and matlab

snonis committed:
Add new flag file

lwpiotr committed:
Merge remote-tracking branch 'origin/dev_sim2root_merge__merge_with_dev_fix_fields' into dev_sim2root_merge__merge_with_dev_fix_fields

lguelzow committed:
new noise trace length

lwpiotr committed:
Added start_event and start_entry to the EventList

lwpiotr committed:
Merge branch 'dev_io_root' into dev_sim2root_merge__merge_with_dev_fix_fields

Commit:
readme

Commit:
New self submitted archiving procedure

lguelzow committed:
Merge branch 'dev_sim2root_merge__merge_with_dev_fix_fields' of https://github.com/grand-mother/grand into dev_sim2root_merge__merge_with_dev_fix_fields

lwpiotr committed:
Speeedups and memory use reduction

lwpiotr committed:
Event now loads (sim)shower if available

lwpiotr committed:
Merge branch 'dev_io_root' into dev_sim2root_merge__merge_with_dev_fix_fields

lwpiotr committed:
Merge remote-tracking branch 'origin/dev_sim2root_merge__merge_with_dev_fix_fields' into dev_sim2root_merge__merge_with_dev_fix_fields

# Conflicts:
#	scripts/convert_voltage2adc.py

lguelzow committed:
remove factor of 2 from trace sample number to get the right number

lwpiotr committed:
Fixed the help and plotting

lwpiotr committed:
Moved

lwpiotr committed:
Changed to a new galactic noise file - not working yet!

lwpiotr committed:
Merge remote-tracking branch 'origin/dev_sim2root_merge__merge_with_dev_fix_fields' into dev_sim2root_merge__merge_with_dev_fix_fields

lwpiotr committed:
Moved to old

lwpiotr committed:
Not sure what changed

lwpiotr committed:
Moved

Commit:
Removed event table feed.
Replaced use of path.rglob (which keeps files/dirs open and can lead to run over the system's file descriptor limit in case of loop over many files) by a simple path concatenation.

claireguepin commented on data/readme.md:
```diff
@@ -13,7 +13,7 @@ Automatically loaded with env/setup.sh,
-## How reload a GRAND model
+## How to reload a GRAND model
 
 * remove/change name of directory grand/data/model/detector
```
grand/data/detector

claireguepin commented on data/download_new_RFchain.py:
```diff
@@ -0,0 +1,73 @@
+print("==============================")
+print("Download new RFchain model (~ 450 KB) for GRAND, please wait ...")
+try:
+    request.urlretrieve(LINK_MODEL, tar_file)
```
`tar_file, headers = request.urlretrieve(LINK_MODEL)` instead of `request.urlretrieve(LINK_MODEL, tar_file)`

claireguepin commented on data/download_data_grand.py:
```diff
@@ -34,7 +32,7 @@
-print("Download data model (~ 1GB) for GRAND, please wait ...")
+print("Download data model (~ 1.3 GB) for GRAND, please wait ...")
 try:
     request.urlretrieve(LINK_MODEL, tar_file)
```
With docker, for this script to work, I need
`tar_file, headers = request.urlretrieve(LINK_MODEL)`
instead of
`request.urlretrieve(LINK_MODEL, tar_file)`

lguelzow committed:
shift antenna z coordinate to reflect shift from obs_lebel/altitude instead of sea level

lguelzow committed:
Merge branch 'dev_sim2root_merge__merge_with_dev_fix_fields' of https://github.com/grand-mother/grand into dev_sim2root_merge__merge_with_dev_fix_fields

claireguepin commented on examples/aoi/data_play.ipynb:
Warning 
```
/home/grand/grand/aoi/event.py:878: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.
  t_vectors = np.array(t_vectors)
```

claireguepin commented on examples/aoi/browse_gp13_events_example.py:
does not run with all file formats
more general: problems of display with docker

claireguepin commented on examples/aoi/dummy_reconstruction.ipynb:
again warning
```
/home/grand/grand/aoi/event.py:878: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.
  t_vectors = np.array(t_vectors)
```

claireguepin commented on examples/aoi/dummy_reconstruction.ipynb:
Concerning
`print(e.simshower.energy_em, e.simshower.Xmax, e.simshower.azimuth, e.simshower.zenith)`
is there a plan to add units? Is it defined somewhere in the code?

claireguepin commented on examples/aoi/browse_sim2root_events_example.py:
similarly to browse_gp13..., pb of display with docker

claireguepin reviewed (commented):
Reviewed directories: data, env, and examples/aoi, with limits related to display and docker (Jupyter server works).

claireguepin commented on examples/dataio/data_storing.py:
Unsure about the difference between this script and aoi/event_generation.py. The script aoi/data_play.py can also load the produced stored_data.root

claireguepin commented on examples/dataio/data_reading.py:
Similar question than for data_storing.py, what is the diff of purpose between this script and aoi/data_play.py?

claireguepin commented on examples/dataio/datafile_use.py:
This file is not described in readme.md. It would be useful to explain the usage and the difference with data_reading.py etc.

claireguepin commented on examples/dataio/ioroot_3dtraces.py:
script not described in readme.md
what types of inputs are we supposed to give?
does not work with stored_data.root, or with old GP13 data, GP80 or GP300 files

claireguepin commented on examples/datalib/datamanager_example.py:
First attempt
```
12:26:14.834 ERROR [granddb.granddata 95] Incoming directory /home/examples/datalib/incoming/ does not exists.
12:26:14.834 ERROR [granddb.granddata 95] Incoming directory /home/examples/datalib/incoming/ does not exists.
```

claireguepin replied:
Need to create directory by hand, this could be easily corrected

claireguepin replied:
And readme.md missing here -> could be useful to indicate what needs to be changed in config.ini so that the script works, and what is the downloaded file Coarse3.root

claireguepin commented on examples/old/grandlib_introduction_v1.ipynb:
```
File /home/grand/scripts/test_voltage_old_rfchain_newLeff_Voc.root doesn't exist.
Traceback (most recent call last):
  File "plot_test.py", line 24, in <module>
    Voc_1 = get_file_event("/home/grand/scripts/test_voltage_old_rfchain_newLeff_Voc.root")
  File "/home/grand/grand/dataio/root_files.py", line 208, in get_file_event
    raise FileNotFoundError
FileNotFoundError

```

claireguepin commented on examples/jm_rootfiles_based/class_Handling3dTraces.ipynb:
Files are missing for the part involving GRANDROOT files (FileNotFoundError)
```
[19]: f_root  = "../../sim2root/Common/sim_Xiaodushan_20221026_180000_RUN0_CD_DC2Alpha_0000/efield_0-1_L0_0000.root"
ef3d = get_handling3dtraces(f_root)
```

claireguepin replied:
Changing to local file for testing gives
```
The branch site was not found in the source file and will not be filled.
The branch site was not found in the source file and will not be filled.
```
But the visuals seem to be produced correctly

claireguepin reviewed (commented):
Reviewed directories: 
- examples/dataio
- examples/datalib
- examples/jm_rootfiles_based

claireguepin commented on examples/jm_rootfiles_based/plot_test.py:
```
File /home/grand/scripts/test_voltage_old_rfchain_newLeff_Voc.root doesn't exist.
Traceback (most recent call last):
  File "plot_test.py", line 24, in <module>
    Voc_1 = get_file_event("/home/grand/scripts/test_voltage_old_rfchain_newLeff_Voc.root")
  File "/home/grand/grand/dataio/root_files.py", line 208, in get_file_event
    raise FileNotFoundError
FileNotFoundError
```

claireguepin commented on examples/sim/antenna.ipynb:
runs, lacks some commenting

claireguepin commented on examples/sim/shower_event.ipynb:
error with [5]:
```
---------------------------------------------------------------------------
MissingFrameError                         Traceback (most recent call last)
Cell In[5], line 59
     57 print("compute_voltage")
     58 # Xmax, Efield, and input frame are all in shower frame.
---> 59 voltage = antenna.compute_voltage(shower.maximum, efield_idx, shower.frame)
     60 print(f"\nVpp= {max(voltage.V) - min(voltage.V)}")
     62 plt.figure()

File /home/grand/grand/sim/detector/process_ant.py:281, in AntennaProcessing.compute_voltage(self, xmax, efield, frame)
    279     print("pos",self.pos)
    280     print("frame",frame)
--> 281     raise MissingFrameError("missing antenna or shower frame")
    283 # Compute the voltage. input fft_leff and field are in shower frame.
    284 leff_f = self.effective_length(xmax, efield, frame)

MissingFrameError: missing antenna or shower frame


```

claireguepin replied:
in particular, the prints give
```
---------------------------------
compute_voltage
pos [[nan]
 [nan]
 [nan]]
frame [[ 3.40202361e-10]
 [-1.18743144e-10]
 [ 3.17104684e-10]]
```

claireguepin commented on examples/sim/shower_event.py:
in general this directory lacks a general description with a readme.md

claireguepin commented on sim2root/ComandsToActualizeExamples.txt:
this file should be merged with the readme.md

claireguepin commented on tests/basis/test_signal.py:
General comment for all test files
```
================================================== short test summary info ===================================================
FAILED tests/basis/test_pipeline.py::PipelineTest::test_add - AttributeError: 'TShower' object has no attribute 'get_run'
FAILED tests/basis/test_signal.py::test_get_peakamptime_norm_hilbert - assert array([512]) == 511
FAILED tests/dataio/test_root_files.py::RootFilesTest::test_FileEfield - IndexError: list index out of range
FAILED tests/dataio/test_root_files.py::RootFilesTest::test_FileVoltage - AssertionError: False is not true
FAILED tests/dataio/test_root_files.py::RootFilesTest::test_fileeventbase - TypeError: __init__() missing 1 required positional argument: 'f_name'
FAILED tests/dataio/test_root_files.py::RootFilesTest::test_get_file_event1 - AssertionError: False is not true
FAILED tests/dataio/test_root_files.py::RootFilesTest::test_get_file_event2 - IndexError: list index out of range
FAILED tests/dataio/test_root_files.py::test_get_obj_handling3dtraces - IndexError: list index out of range
FAILED tests/dataio/test_root_trees.py::RootTreesTest::test_datadirectory - IndexError: list index out of range
FAILED tests/dataio/test_root_trees.py::RootTreesTest::test_tadc - AssertionError: False is not true
FAILED tests/dataio/test_root_trees.py::RootTreesTest::test_trunshowersim - AssertionError: False is not true
FAILED tests/dataio/test_root_trees.py::RootTreesTest::test_trunvoltage - AssertionError: False is not true
FAILED tests/dataio/test_root_trees.py::RootTreesTest::test_tshowersim - AssertionError: False is not true
FAILED tests/geo/test_topography.py::TopographyTest::test_custom_topography - grand.geo.turtle.LibraryError: A TURTLE library error occurred: b"{ turtle_stack_create [#3], src/turtle/io/hgt.c:82 } in...
FAILED tests/geo/test_topography.py::TopographyTest::test_topography_cache - AssertionError: [PosixPath('/home/grand/data/topography/._N41E096.hgt'), PosixPath('/home/grand/data/topography/N41E096.h...
FAILED tests/geo/test_topography.py::TopographyTest::test_topography_distance - grand.geo.turtle.LibraryError: A TURTLE library error occurred: b"{ turtle_stack_create [#3], src/turtle/io/hgt.c:82 } in...
FAILED tests/geo/test_topography.py::TopographyTest::test_topography_elevation - grand.geo.turtle.LibraryError: A TURTLE library error occurred: b"{ turtle_stack_create [#3], src/turtle/io/hgt.c:82 } in...
FAILED tests/geo/test_turtle.py::TurtleTest::test_stack - grand.geo.turtle.LibraryError: A TURTLE library error occurred: b"{ turtle_stack_create [#3], src/turtle/io/hgt.c:82 } in...
FAILED tests/geo/test_turtle.py::TurtleTest::test_stepper - grand.geo.turtle.LibraryError: A TURTLE library error occurred: b"{ turtle_stack_create [#3], src/turtle/io/hgt.c:82 } in...
FAILED tests/sim/test_antenna.py::AntennaTest::test_antenna - TypeError: ufunc 'isfinite' not supported for the input types, and the inputs could not be safely coerced to any supporte...
FAILED tests/sim/test_efield2voltage.py::Efield2VoltageTest::test_Efield2Voltage - AttributeError: 'TShower' object has no attribute 'get_run'

```

claireguepin replied:
```
21 failed, 69 passed, 1 warning in 14.58s
```

claireguepin commented on sim2root/ZHAireSRawRoot/README.MD:
not sure this is necessary as already described in the general readme.md

claireguepin commented on sim2root/README.md:
```diff
@@ -38,11 +38,97 @@ Alternatively, you can specify any other directory including a full CoREAS simul
+
+i.e.
+
+`python3 python  ../grand/sim2root/Common/sim2root.py <your path>*/*.rawroot -d 20221026 -t 180000 -e DC2Alpha`
```
This doesn't work. First issues: remove `python`, and add `-sl GP300`

claireguepin commented on sim2root/Common/sim2root.py:
When trying to run
```
python3 ./sim2root.py ./../ZHAireSRawRoot/GP300_Xi_Sib_Proton_3.8_51.6_135.4_1618.rawroot -d 20221026 -t 180000 -e DC2Alpha -sl GP300
```
I have the following problem
```
13:48:01.243  INFO [grand.manage_log 187] create handler for root logger: ['grand']
Traceback (most recent call last):
  File "./sim2root.py", line 980, in <module>
    main()
  File "./sim2root.py", line 302, in main
    out_dir_name = init_all_trees(clargs, trawshower.unix_date, run_number, site, gt)
  File "./sim2root.py", line 526, in init_all_trees
    date,time=datetime.datetime.fromtimestamp(unix_date, datetime.UTC).strftime('%Y%m%d_%H%M%S').split("_")
AttributeError: module 'datetime' has no attribute 'UTC'

```

claireguepin replied:
Changing to `datetime.timezone.utc`
Maybe it's a version's issue

claireguepin replied:
Then another issue. Output:
```
13:52:38.993  INFO [grand.manage_log 187] create handler for root logger: ['grand']
13:52:39.505 WARNING [grand.dataio.data_tree 391] No valid trun TTree in the file sim_Xiaodushan_20221026_180000_RUN1_CD_DC2Alpha_0000/run.root. Creating a new one.
13:52:39.589 WARNING [grand.dataio.data_tree 391] No valid trunshowersim TTree in the file sim_Xiaodushan_20221026_180000_RUN1_CD_DC2Alpha_0000/runshowersim.root. Creating a new one.
13:52:39.598 WARNING [grand.dataio.data_tree 391] No valid trunefieldsim TTree in the file sim_Xiaodushan_20221026_180000_RUN1_CD_DC2Alpha_0000/runefieldsim.root. Creating a new one.
13:52:39.602 WARNING [grand.dataio.data_tree 391] No valid tshower TTree in the file sim_Xiaodushan_20221026_180000_RUN1_CD_DC2Alpha_0000/shower.root. Creating a new one.
13:52:39.607 WARNING [grand.dataio.data_tree 391] No valid tshowersim TTree in the file sim_Xiaodushan_20221026_180000_RUN1_CD_DC2Alpha_0000/showersim.root. Creating a new one.
13:52:39.742 WARNING [grand.dataio.data_tree 391] No valid tefield TTree in the file sim_Xiaodushan_20221026_180000_RUN1_CD_DC2Alpha_0000/efield.root. Creating a new one.
Traceback (most recent call last):
  File "./sim2root.py", line 980, in <module>
    main()
  File "./sim2root.py", line 339, in main
    rawshower2grandroot(trawshower, gt)
  File "./sim2root.py", line 700, in rawshower2grandroot
    gt.tshower.energy_em = trawshower.energy_em[0]
  File "/home/grand/grand/dataio/descriptors.py", line 86, in __getitem__
    raise IndexError("list index out of range")
IndexError: list index out of range
```

claireguepin replied:
The trawshower.energy_em is an empty list
After commenting this line, I have the following issue
```
Traceback (most recent call last):
  File "./sim2root.py", line 980, in <module>
    main()
  File "./sim2root.py", line 346, in main
    trace = np.stack([trawefield.trace_x, trawefield.trace_y, trawefield.trace_z], 1, dtype=np.float32)
  File "<__array_function__ internals>", line 179, in stack
TypeError: _stack_dispatcher() got an unexpected keyword argument 'dtype'
```
I suspect a version problem

claireguepin commented on sim2root/Common/RunSimPipe.py:
When trying 
```
python RunSimPipe.py ../ZHAireSRawRoot ZHAireS
```
output
```
DEBUG:root:PYTHONINTERPRETER not defined, defaulting to python
DEBUG:root: Trying to make GrandRoot file
about to run:python ./sim2root.py ../ZHAireSRawRoot --target_duration_us=4.096 --trigger_time_ns 800 -e ZHAireS
b"14:00:19.192  INFO [grand.manage_log 187] create handler for root logger: ['grand']\n"
DEBUG:root: Trying to produce voltages
Traceback (most recent call last):
  File "RunSimPipe.py", line 61, in <module>
    OUTPUTFILE=OUTPUTFILE[0].replace("efield", "voltage")
IndexError: list index out of range
```

claireguepin reviewed (commented):
Some attempts to run
- pytest tests
- several examples in sim2root

claireguepin commented on scripts/readme.md:
only 4 out of 12 scripts are described in this readme
it would be useful to add more info or comment scripts more thoroughly

Commit:
Changes to use dataio.
New dataset registration procedure.
Simplifications.
This version need more testing (and can break older procedures).

Commit:
Changes to use dataio.
New dataset registration procedure.
Simplifications.
This version need more testing (and can break older procedures).

Commit:
Merge remote-tracking branch 'origin/dev_io_root' into dev_database

# Conflicts:
#	data/download_data_grand.py
#	examples/sim/shower_event.ipynb
#	examples/sim/shower_event.py
#	grand/sim/detector/rf_chain.py
#	grand/sim/efield2voltage.py
#	granddb/datamanager.py
#	granddb/granddblib.py
#	scripts/convert_efield2voltage.py
#	scripts/convert_voltage2adc.py
#	sim2root/Common/IllustrateSimPipe.py
#	sim2root/Common/sim2root.py
#	tests/sim/test_shower.py

claireguepin commented on scripts/convert_efield2efield.py:
Tested with sim_Xiaodushan_20221026_180000_RUN1_CD_DC2Alpha_0000, crashed:
```
Traceback (most recent call last):
  File "convert_efield2efield.py", line 268, in <module>
    d_input = grand.dataio.DataDirectory(args.directory)
  File "/home/grand/grand/dataio/data_handling.py", line 36, in __init__
    self.file_handle_list = self.get_list_of_files_handles()
  File "/home/grand/grand/dataio/data_handling.py", line 76, in get_list_of_files_handles
    for key, filenames in groupby(sorted(self.file_list, key=split_filenames), split_filenames):
  File "/home/grand/grand/dataio/data_handling.py", line 72, in split_filenames
    return el[0], el[2]
IndexError: list index out of range
```

Tested with sim_Xiaodushan_20221026_000000_RUN1_CD_ZHAireS_0000, crashed:
```
Traceback (most recent call last):
  File "convert_efield2efield.py", line 592, in <module>
    out_tefield.fill()
  File "/home/grand/grand/dataio/event_trees.py", line 43, in fill
    raise NotUniqueEvent(
grand.dataio.descriptors.NotUniqueEvent: An event with (run_number,event_number)=(1,13790) already exists in the TTree tefield.
```

claireguepin commented on scripts/convert_efield2voltage.py:
Tested with sim_Xiaodushan_20221026_180000_RUN1_CD_DC2Alpha_0000/efield.root, crashed:
```
Traceback (most recent call last):
  File "convert_efield2voltage.py", line 202, in <module>
    signal = Efield2Voltage(args.directory, args.out_file, output_directory=args.out_directory, seed=seed, padding_factor=args.padding_factor, du_type=args.du_type)
  File "/home/grand/grand/sim/efield2voltage.py", line 67, in __init__
    f_input_TRun = self.d_input.trun
AttributeError: 'DataFile' object has no attribute 'trun'
```

Same with sim_Xiaodushan_20221026_000000_RUN1_CD_ZHAireS_0000/efield_1618-13790_L0_0000.root

claireguepin replied:
Also tested test_efield.root, and crashed:
```
11:04:37.122  INFO [grand.sim.efield2voltage 129] Running on event_number: 1, run_number: 0
11:04:37.152  INFO [grand.dataio.data_tree 683] Could not find trigger_position in tree tefield. This field won't be assigned.
Traceback (most recent call last):
  File "convert_efield2voltage.py", line 214, in <module>
    signal.compute_voltage()    # saves automatically
  File "/home/grand/grand/sim/efield2voltage.py", line 572, in compute_voltage
    self.compute_voltage_event(event_idx=evt_idx) # event_number and run_number is None
  File "/home/grand/grand/sim/efield2voltage.py", line 496, in compute_voltage_event
    self.compute_voc_event(event_idx, event_number, run_number)
  File "/home/grand/grand/sim/efield2voltage.py", line 416, in compute_voc_event
    self.get_event(event_idx, event_number, run_number)
  File "/home/grand/grand/sim/efield2voltage.py", line 134, in get_event
    self.run.get_run(self.run_number)                         # update run info to get site latitude and longitude.
AttributeError: 'TShower' object has no attribute 'get_run'
```

claireguepin commented on scripts/convert_voltage2adc.py:
Tested with sim_Xiaodushan_20221026_000000_RUN1_CD_ZHAireS_0000, crashed:
```
11:02:08.196  INFO [convert_voltage2adc 301] Converting voltage to ADC for entry 1/2
Traceback (most recent call last):
  File "convert_voltage2adc.py", line 358, in <module>
    tadc.fill()
  File "/home/grand/grand/dataio/event_trees.py", line 43, in fill
    raise NotUniqueEvent(
grand.dataio.descriptors.NotUniqueEvent: An event with (run_number,event_number)=(1,13790) already exists in the TTree tadc.
```

claireguepin commented on scripts/open_grand_analysis_prompt.py:
```
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
Cell In[1], line 1
----> 1 from grand.grandlib_classes.grandlib_classes import *; el = EventList('data_test/sim_Xiaodushan_20221026_000000_RUN1_CD_ZHAireS_0000/'); print(f'\n\033[0;31mCreated a list of events in directory data_test/sim_Xiaodushan_20221026_000000_RUN1_CD_ZHAireS_0000/ as el\033[0m\n'); print('You can now iterate through events with, for example:\n\nfor i,e in enumerate(el):\n  print(e.event_number)\n  ...')

ModuleNotFoundError: No module named 'grand.grandlib_classes.grandlib_classes'

```

claireguepin commented on scripts/plot_tmax_vmax.py:
this script lacks a description

claireguepin commented on granddb/rootdblib.py:
several files lack a general description of their purpose, this could be useful to add eventually

snonis committed:
Updated /examples/scripts to use PG_ALL_jifen.mat instead of 30_250galactic.mat

Commit:
test

Commit:
test

Commit:
Code cleaning

Commit:
No change

lwpiotr committed:
Added comment about a possible bug

lwpiotr committed:
Moved

lwpiotr committed:
Removed explicit conversion to string

lwpiotr committed:
A fix for numpy version >2

lwpiotr committed:
Setting directory for trawvoltage only files; Workaround for missing proper DUs in trun for GP300

lwpiotr committed:
Improved checking for trees in the data

lwpiotr committed:
Merge branch 'dev_sim2root_merge__merge_with_dev_fix_fields' into dev_io_root

lwpiotr committed:
Fixed the GP300 antenna workaround for specific entry request

lguelzow committed:
rename adc entries as they affected a logging output for the voltage entry number

Commit:
Put path to createAIP.jar into var in archive_grandraw.bash

Commit:
Revert "Removed all import from grand/__init__.py and adjusted imports in other files when needed"

This reverts commit 09eaef0b

Commit:
Merge remote-tracking branch 'origin/dev_sim2root_merge__merge_with_dev_fix_fields' into dev_database

Commit:
Merge remote-tracking branch 'origin/dev_sim2root_merge__merge_with_dev_fix_fields' into dev_database

# Conflicts:
#	examples/old/grandlib_introduction_v1.ipynb

Commit:
restore modified files from dev_sim2root_merge__merge_with_dev_fix_fields

Commit:
restore modified files from dev_sim2root_merge__merge_with_dev_fix_fields

Commit:
Added path detection of the current script to call other scripts in the same directoryin archive_grandraw.bash.
Added better management of directories when gtot uses the new format in bintoroot.bash.

Commit:
Merge remote-tracking branch 'origin/dev_database' into dev_database

# Conflicts:
#	examples/old/grandlib_introduction_v1.ipynb
#	scripts/transfers/bintoroot.bash

lwpiotr committed:
Added universal function for initialisation of DataDirectory structure: init_structure()

lwpiotr committed:
DataDirectory: Added a tree_types variable; Added close() to close all the DataFiles; moved tree_instances to the class init

lwpiotr committed:
Moved DataFile values to init, to avoid sharing between instances

lwpiotr committed:
Made working with ROOT 6.36 and fixing the case when char is converted into short

lwpiotr committed:
Deleting in-memory histograms that appear after Draw

lwpiotr committed:
A script for extracting events from provided directories and storing them in a target directory

lwpiotr committed:
Added an option to overwrite the target directory

luckyjim committed:
add max in plot event viewer

luckyjim committed:
Merge remote-tracking branch 'origin/dev_fix_dc2' into dev_sim2root_merge__merge_with_dev_fix_fields

lwpiotr committed:
Generating indices for files

lwpiotr committed:
Removed some debug output

lwpiotr committed:
Merge remote-tracking branch 'origin/dev_sim2root_merge__merge_with_dev_fix_fields' into dev_sim2root_merge__merge_with_dev_fix_fields

lwpiotr committed:
Merge branch 'dev_io_root' into dev_sim2root_merge__merge_with_dev_fix_fields

lwpiotr committed:
Added GP13 to antennas positions workaround

lwpiotr committed:
Fixed an import; Added option to use trawvoltage

lwpiotr committed:
Merge branch 'dev_io_root' into dev_sim2root_merge__merge_with_dev_fix_fields

lwpiotr committed:
Merge remote-tracking branch 'origin/dev_snonis' into dev_snonis

lwpiotr committed:
Merge branch 'dev_snonis' into dev_sim2root_merge__merge_with_dev_fix_fields

# Conflicts:
#	data/download_data_grand.py
#	examples/sim/rf_chain_example.py

lwpiotr committed:
DataFile: fix max_tree_instance in case of no analysis_level

lwpiotr committed:
Changed to a stub informing which working script to check.

lwpiotr committed:
Merge branch 'dev_io_root' into dev_sim2root_merge__merge_with_dev_fix_fields

lwpiotr committed:
Changed to a stub informing which working script to check.
